### PR TITLE
fix:[CORE-1495] Permission issue for webapp

### PR DESCRIPTION
**Pull request by AishwaryaKulkarni97**
# Description

Ops alert should not be raised for webapp AuthorisationFailed error

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Chore (no code changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My commit message/PR follows the contribution guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java [hunk 1]
```diff
@@ -28,10 +29,9 @@ public List<WebAppVH> fetchWebAppDetails(SubscriptionVH subscription) {
         Azure azure = azureCredentialProvider.getClient(subscription.getTenant(), subscription.getSubscriptionId());
 
         PagedList<WebApp> webApps = azure.webApps().list();
-
+        WebAppVH webAppVH = new WebAppVH();
```
Comment (plyubich) [inline jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java:32]: Why move it here?

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java [hunk 2]
```diff
@@ -28,10 +29,9 @@ public List<WebAppVH> fetchWebAppDetails(SubscriptionVH subscription) {
...
         for (WebApp webApp : webApps) {
             try {
-                WebAppVH webAppVH = new WebAppVH();
```
Comment (plyubich) [inline jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java:34]: Can we have a contractor that simply takes 3 params?

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/ErrorManageUtil.java [hunk 3]
```diff
@@ -144,7 +144,7 @@ public static void triggerNotificationforPermissionDenied()
 			List<ErrorVH> errorVHList=entry.getValue();
 			for(ErrorVH errorVH:entry.getValue())
 			{
-				if(errorVH.getType().equals("vault")&&errorVH.getException().contains("DeniedWithNoValidRBAC"))
+				if((errorVH.getType().equals("vault")&&errorVH.getException().contains("DeniedWithNoValidRBAC"))||(errorVH.getType().equals("webapp")&&errorVH.getException().contains("AuthorizationFailed")))
```
Comment (plyubich) [inline jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/ErrorManageUtil.java:147]: Reformat file please

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java [hunk 4]
```diff
@@ -4,6 +4,7 @@
 import com.microsoft.azure.management.Azure;
 import com.microsoft.azure.management.appservice.DefaultErrorResponseException;
 import com.microsoft.azure.management.appservice.WebApp;
+import com.tmobile.pacbot.azure.inventory.ErrorManageUtil;
```
Comment (plyubich) [inline jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java:7]: Reformat file please using ide format file

**Review by plyubich — CHANGES_REQUESTED** [review]

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java (hunk 1 above)
Comment (AishwaryaKulkarni97) [reply jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java:32]: Need to get region and subscription in ->  ErrorManageUtil.uploadError(webAppVH.getSubscription(),webAppVH.getRegion(),"webapp",exception.getMessage());

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/ErrorManageUtil.java (hunk 3 above)
Comment (AishwaryaKulkarni97) [reply jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/ErrorManageUtil.java:147]: Addressed in commit

File: jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java (hunk 4 above)
Comment (AishwaryaKulkarni97) [reply jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/WebAppInventoryCollector.java:7]: Addressed in commit

**Review by santhosh-challa — APPROVED** [review]
LGTM

**Pull request merged**